framework/src/modules/keys/register_asset.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable class-methods-use-this */
+import { getAddressFromPublicKey } from '@liskhq/lisk-cryptography';
+import { BaseAsset, ApplyAssetInput, StateStore, ValidateAssetInput } from '../base_asset';
+import { KeysSchema } from './schemas';
+import { AccountKeyAsset } from './types';
+
+interface Asset {
+	mandatoryKeys: Array<Readonly<Buffer>>;
+	optionalKeys: Array<Readonly<Buffer>>;
+	readonly numberOfSignatures: number;
+}
+
+const setMemberAccounts = async (
+	store: StateStore,
+	membersPublicKeys: Array<Readonly<Buffer>>,
+): Promise<void> => {
+	for (const memberPublicKey of membersPublicKeys) {
+		const address = getAddressFromPublicKey(memberPublicKey as Buffer);
+		// Key might not exists in the blockchain yet so we fetch or default
+		const memberAccount = await store.account.getOrDefault(address);
+		store.account.set(memberAccount.address, memberAccount);
+	}
+};
+
+const hasDuplicateKey = (keys: Buffer[]): boolean => {
```

Comment:
Not important, but can be replaced by `bufferArrayUniqueItems`

Reply:
Done.